lib/promscrape/targetstatus.qtpl

```diff
@@ -170,19 +170,19 @@
         <div class="tab-pane active" role="tabpanel">
             {% switch activeTab %}
             {% case "scrapeTargets" %}
-                {%= scrapeTargets(tsr) %}
+                {%= scrapeTargets(tsr, filter) %}
```

Comment:
wdyt about filtering empty jobs here https://github.com/VictoriaMetrics/VictoriaMetrics/blob/ae4d376e4122bd4c6e542400406ba19bab0051f1/lib/promscrape/targetstatus.go#L463-L490 instead of propagating filters in templates?

Reply:
```
@@ -475,6 +475,9 @@ func (tsm *targetStatusMap) getTargetsStatusByJob(filter *requestFilter) *target
                        }
                        targetsStatuses = append(targetsStatuses, ts)
                }
+               if filter.showOnlyUnhealthy && len(targetsStatuses) == 0 {
+                       continue
+               }
                jts = append(jts, &jobTargetsStatuses{
                        jobName:       jobName,
                        upCount:       ups,
@@ -488,7 +491,7 @@ func (tsm *targetStatusMap) getTargetsStatusByJob(filter *requestFilter) *target
        emptyJobs := getEmptyJobs(jts, jobNames)
        var err error
        jts, err = filterTargets(jts, filter.endpointSearch, filter.labelSearch)
-       if len(filter.endpointSearch) > 0 || len(filter.labelSearch) > 0 {
+       if len(filter.endpointSearch) > 0 || len(filter.labelSearch) > 0 || filter.showOnlyUnhealthy {
                // Do not show empty jobs if target filters are set.
                emptyJobs = nil
        }

```

Reply:
I will, update the logic and check. Thank you

Reply:
@hagen1778 , I have checked the logic. It works the same way as using filters in the template, but the code is simpler. Tank you for your recommendation 👍🏻